sharedbuffers/mapped_struct.py

```diff
@@ -1547,6 +1588,22 @@ def __init__(self, buf, offset = 0, idmap = None, idmap_size = 1024):
             count = self.index_elements)
         self.idmap = idmap
 
+        if self.index_elements > 0 and self.index[0] >= (self._Header.size + self._NewHeader.size):
+            # New version, most likely
+            self.version, self.schema_offset, self.schema_size = self._NewHeader.unpack_from(buf, self._Header.size)
+            if self.schema_offset and self.schema_size:
+                if self.schema_offset > len(buf) or (self.schema_size + self.schema_offset) > len(buf):
+                    raise ValueError("Corrupted input - bad schema location")
+                stored_schema = cPickle.loads(bytes(buffer(buf, self.schema_offset, self.schema_size)))
+                if not isinstance(stored_schema, Schema):
+                    raise ValueError("Corrupted input - unrecognizable schema")
+                if self.schema is None or not self.schema.compatible(stored_schema):
+                    self.schema = stored_schema
+            elif self.schema is None:
+                raise ValueError("Cannot map schema-less buffer without specifying schema")
+        else:
+            self.version = 0
```

Comment:
I was talking about the version field in the on-disk format.

The attribute also makes sense, to have it for future code if it needs to know the version somehow. Note that future code could be external code. So it's an interface added because it seems useful, even if we make no immediate use of it.